packages/cli/src/hooks/analytics/session.spec.ts

```diff
@@ -0,0 +1,117 @@
+jest.mock('../../lib/config/config');
+
+import type {IConfig} from '@oclif/config';
+import {mocked} from 'ts-jest/utils';
+import {Config} from '../../lib/config/config';
+import check from './session';
+
+describe('session', () => {
+  const mockedConfig = mocked(Config, true);
+  const mockedConfigGet = jest.fn();
+  const mockedConfigSet = jest.fn();
+  const theDate = new Date('1997-08-29T12:00:00Z');
+  const dummySessionId = 1234567;
+
+  const doMockConfig = () => {
+    mockedConfig.prototype.get = mockedConfigGet;
+    mockedConfig.prototype.set = mockedConfigSet;
+  };
+
+  const freezeTime = () => {
+    jest.useFakeTimers().setSystemTime(theDate.getTime());
+  };
+
+  const getTimeBefore = (minutes: number) => {
+    return theDate.getTime() - minutes * 60e3;
```

Comment:
feel free to ignore, but, you can use: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat to formulate duration/elapsed time in a more human-friendly manner (begone multiplication! Begone! 😂 )

Same apply to the `session.ts` file ;)

Reply:
Nice, I did not know about that... but I don't see how I can use RelativeTimeFormat to perform compare dates

Reply:
Hm you right, there'd be ways by playing with the rest of Intl... But that's not the goal of this library thus we may hit hiccups. Ignore this thread 😂